src/components/NewCommentForm.tsx

```diff
@@ -75,29 +169,52 @@ export const NewCommentForm: React.FC = () => {
             id="comment-body"
             name="body"
             placeholder="Type comment here"
-            className="textarea is-danger"
+            className={classNames(
+              'textarea', { 'is-danger': bodyError },
+            )}
+            value={newCommentsData.body}
+            onChange={inputCommentsBody}
           />
         </div>
 
-        <p className="help is-danger" data-cy="ErrorMessage">
-          Enter some text
-        </p>
+        {bodyError && (
+          <p className="help is-danger" data-cy="ErrorMessage">
+            Enter some text
+          </p>
+        )}
       </div>
 
       <div className="field is-grouped">
         <div className="control">
-          <button type="submit" className="button is-link is-loading">
+          <button
+            type="submit"
+            className={classNames(
+              'button',
+              'is-link',
+              { 'is-loading': isCommentAdding },
+            )}
+            onClick={handleAddClick()}
```

Comment:
Remove the parentheses after 'handleAddClick' since it should be passed as a reference, not invoked.